### ISSUE TITLE
FEAT: check if using dot command in posix shell with arguments

### ISSUE THREAD

**Issue by tomelliff** [koalaman/shellcheck#1373]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
. "${PWD}"/test-child.sh foo
```

#### Here's what shellcheck currently says:
```
In test-parent.sh line 9:
. "${PWD}"/test-child.sh foo
  ^--------------------^ SC1090: Can't follow non-constant source. Use a directive to specify location.

For more information:
  https://www.shellcheck.net/wiki/SC1090 -- Can't follow non-constant source. Use...
```

#### Here's what I wanted or expected to see:
SC1090: Can't follow non-constant source. Use a directive to specify location.
SC2xxx: Can't specify arguments to a script sourced with dot in POSIX shell

---

POSIX shell always passes the arguments of the called script into the sourced script so if you have something like:

### test-parent.sh
```sh
#!/bin/sh

echo "$0"

echo "parent"

echo "$@"

. "${PWD}"/test-child.sh foo
```
and
### test-child.sh
```sh
#!/bin/sh

echo "$0"

echo "child"

echo "$@"
```
You get the following output:
```
$ ./test-parent.sh foo bar baz
./test-parent.sh
parent
foo bar baz
./test-parent.sh
child
foo bar baz
```
Switching to bash gives:
```
$ ./test-parent.sh foo bar baz
./test-parent.sh
parent
foo bar baz
./test-parent.sh
child
foo
```

**Comment by koalaman** [comment]
I didn't know that! Definitely worth a warning. Thanks!
